lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -120,12 +126,21 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
                 config.getSseKmsKeyId(),
                 Optional.empty(),
                 config.getCannedAcl());
+
+        this.uploadExecutor = new ThreadPoolExecutor(
+                0,
+                config.getStreamingMaxUploadThreads(),
+                60L,
+                TimeUnit.SECONDS,
+                new SynchronousQueue<>(),
+                daemonThreadsNamed("s3-upload-%s"));
     }
 
     @PreDestroy
     public void destroy()
     {
         client.close();
+        uploadExecutor.close();
```

Comment:
`close()` will block up to a day before interrupting tasks.

Since we probably want to use a shorter time frame, we should use `MoreExecutors.shutdownAndAwaitTermination()`, which does the same thing and lets us provide the same time frame. I recommend 10s.

Reply:
Yeah, `close()` was added for structured concurrency with virtual threads. The behavior makes sense for that use case, but not for server shutdown.

Reply:
Good catch, `GcsFileSystemFactory` also uses an executor service so I followed the existing pattern of how it shuts down. `executorService.shutdownNow()` which interrupts each thread and moves on without waiting for termination.

Reply:
@erichwang good point. btw this applies to quite many existing thread pools. 
We will need static code analysis rules to cope with that.

> we should use `MoreExecutors.shutdownAndAwaitTermination()`,

for connector/server shutdown, we should use shutdownNow() without delay, so `shutdownAndAwaitTermination` isn't exactly perfect